src/hyp3_isce2/dem.py

```diff
@@ -74,65 +67,28 @@ def distance_meters_to_degrees(distance_meters, latitude):
     return np.round(distance_degrees_lon, 15), np.round(distance_degrees_lat, 15)
 
 
-def download_dem_for_isce2(
-    extent: list | tuple,
-    dem_name: str = 'glo_30',
-    dem_dir: Path | None = None,
-    buffer: float = 0.4,
-    resample_20m: bool = False,
-) -> Path:
+def download_dem_for_isce2(extent: tuple[float, float, float, float], dem_path: Path, pixel_size: float) -> Path:
     """Download the given DEM for the given extent.
 
     Args:
         extent: A list [xmin, ymin, xmax, ymax] for epsg:4326 (i.e. (x, y) = (lon, lat)).
-        dem_name: One of the names from `dem_stitcher`.
-        dem_dir: The output directory.
-        buffer: The extent buffer in degrees, by default .4, which is about 44 km at the equator
-                (or about 2.5 bursts at the equator).
-        resample_20m: Whether or not the DEM should be resampled to 20 meters.
+        dem_path: The path to write the DEM to.
+        resolution: The resolution of the DEM in meters.
 
     Returns:
         The path to the downloaded DEM.
     """
-    dem_dir = dem_dir or Path()
-    dem_dir.mkdir(exist_ok=True, parents=True)
-
-    extent_buffered = buffer_extent(extent, buffer)
-
-    if resample_20m:
-        res_degrees = distance_meters_to_degrees(20.0, extent_buffered[1])
-        dem_array, dem_profile = dem_stitcher.stitch_dem(
-            extent_buffered,
-            dem_name,
-            dst_ellipsoidal_height=True,
-            dst_area_or_point='Point',
-            n_threads_downloading=5,
-            dst_resolution=res_degrees,
-        )
-        dem_path = dem_dir / 'full_res_geocode.dem.wgs84'
-    else:
-        dem_array, dem_profile = dem_stitcher.stitch_dem(
-            extent_buffered,
-            dem_name,
-            dst_ellipsoidal_height=True,
-            dst_area_or_point='Point',
-            n_threads_downloading=5,
+    with tempfile.NamedTemporaryFile(suffix='.tif') as tmp_dem:
+        prepare_dem_geotiff(
+            tmp_dem,
```

Comment:
hyp3lib.dem.prepare_dem_geotiff expects a `Path`, so I think this should be:
```suggestion
            Path(tmp_dem.name),
```